### PR TITLE
chore: update eth-json-rpc-middleware to ^9

### PR DESCRIPTION
**Pull request by mikesposito**
Fixes [16608](https://github.com/MetaMask/metamask-extension/issues/16608)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
   "dependencies": {
     "@metamask/safe-event-emitter": "^2.0.0",
     "async-mutex": "^0.2.6",
-    "eth-json-rpc-middleware": "^6.0.0",
+    "eth-json-rpc-middleware": "^9.0.0",
```
Comment (legobeat) [inline package.json:23]: This is now only directly imported in `test/util.js` - does that mean this was not actually necessary in `dependencies` and can now be moved to `devDependencies`?

Comment (mikesposito) [reply package.json:23]: Good point.
createScaffoldMiddleware was RE-exported from [eth-json-rpc-middleware@^6](https://github.com/MetaMask/eth-json-rpc-middleware/blob/v6.0.0/scaffold.js#L2).
I think we can just use `json-rpc-engine` directly in this package (as we already import it), moving `eth-json-rpc-middleware` to `devDependencies`

**Review by legobeat — DISMISSED** [review]
lgtm

File: package.json (hunk 1 above)
Comment (legobeat) [reply package.json]: Maybe (most likely) it's totally cool and seems more sane than mixing versions like previously, just noting that this also implies switching `json-rpc-engine` from `5.4.0` to `6.1.0`.

**Comment by legobeat** [comment]
@mikesposito still relevant?

**Review by mcmire — DISMISSED** [review]
This looks good to me as a general housekeeping sort of thing. If there's another reason for making this change (I suspect it may be to make the extension bundle smaller), it would be nice to provide the context in the PR description so we know!

**Comment by Gudahtt** [comment]
Agreed that this looks good!

@mikesposito Could you update this to resolve the lockfile conflict? I will remember to merge it more quickly this time, sorry for the delay.

**Comment by mikesposito** [comment]
> Agreed that this looks good!
> 
> @mikesposito Could you update this to resolve the lockfile conflict? I will remember to merge it more quickly this time, sorry for the delay.

@Gudahtt no problem! 
done, thanks!

**Comment by Gudahtt** [comment]
@SocketSecurity ignore @ethereumjs/rlp@4.0.0 rlp@2.2.6

This bin script is not used by us

**Review by Gudahtt — APPROVED** [review]

**Pull request merged**